### DISCUSSION TITLE
Error in rowMean(m):'x'必须是阵列，而且至少得有两个维度

### DISCUSSION THREAD

**Discussion by liuyu988** [jokergoo/ComplexHeatmap#859] [Q&A]
Hi,guys!I am running the circos.heatmap(),i met some questions.
my code is all from the protocal in R help()

> set.seed(123)
> mat1 = rbind(cbind(matrix(rnorm(50*5, mean = 1), nr = 50),  matrix(rnorm(50*5, mean = -1), nr = 50)),
                cbind(matrix(rnorm(50*5, mean = -1), nr = 50), matrix(rnorm(50*5, mean = 1), nr = 50)))
> rownames(mat1) = paste0("R", 1:100)
> colnames(mat1) = paste0("C", 1:10)
> mat1 =  mat1[sample(100, 100), ] # randomly permute rows
> split = sample(letters[1:5], 100, replace = TRUE)
> spilt = factor(split, levels = letters[1:5])
> col_fun1 = colorRamp2(c(-2, 0, 2), c("blue", "white", "red"))
> circos.heatmap(mat1, split = split, col = col_fun1)

Error in rowMean(m):'x'必须是阵列，而且至少得有两个维度!

Then, i see:
> dim(mat1)
> rowMeans(mat1)

the two functions can run correctly, so I can't understand how to solve the error above  and  why does the error appear?
thanks a lot!

**Comment by jokergoo** [comment]
When you construct `mat1`:

```r
mat1 = rbind(cbind(matrix(rnorm(505, mean = 1), nr = 50), matrix(rnorm(505, mean = -1), nr = 50)),
cbind(matrix(rnorm(505, mean = -1), nr = 50), matrix(rnorm(505, mean = 1), nr = 50)))
```

I think the correct code should be:

```r
at1 = rbind(cbind(matrix(rnorm(250, mean = 1), nr = 50), matrix(rnorm(250, mean = -1), nr = 50)),
            cbind(matrix(rnorm(250, mean = -1), nr = 50), matrix(rnorm(250, mean = 1), nr = 50)))
```

The number for `rnorm()` should be 250 while not 505, right?

![image](https://user-images.githubusercontent.com/449218/152325588-83a908c2-68c9-44bb-abfe-db6d4a2b0105.png)
